teuthology/suite/test/test_merge.py

```diff
@@ -0,0 +1,232 @@
+import logging
+
+from mock import patch, MagicMock
+from unittest import TestCase
+
+from teuthology.suite import build_matrix
+from teuthology.suite.merge import config_merge
+from teuthology.test.fake_fs import make_fake_fstools
+
+log = logging.getLogger(__name__)
+
+class TestMerge(TestCase):
+    patchpoints = [
+        'os.path.exists',
+        'os.listdir',
+        'os.path.isfile',
+        'os.path.isdir',
+        'builtins.open',
+    ]
+
+    def setUp(self):
+        log.debug("setUp")
+        self.mocks = dict()
+        self.patchers = dict()
+        for ppoint in self.__class__.patchpoints:
+            self.mocks[ppoint] = MagicMock()
+            self.patchers[ppoint] = patch(ppoint, self.mocks[ppoint])
+
+    def start_patchers(self, fake_fs):
+        fake_fns = make_fake_fstools(fake_fs)
+        # N.B.: relies on fake_fns being in same order as patchpoints
+        for ppoint, fn in zip(self.__class__.patchpoints, fake_fns):
+            self.mocks[ppoint].side_effect = fn
+            self.patchers[ppoint].start()
+
+    def stop_patchers(self):
+        for patcher in self.patchers.values():
+            patcher.stop()
+
+    def tearDown(self):
+        log.debug("tearDown")
+        self.patchers.clear()
+        self.mocks.clear()
+
+    def test_premerge(self):
+        fake_fs = {
+            'd0_0': {
+                '%': None,
+                'd1_0': {
+                  'a.yaml': """
```

Comment:
Not a big deal, but `textwrap.dedent()` might be useful in cases like this

Reply:
I'll give it a shot, thanks.